docs/electrical/embedded/styleguide.md

```diff
@@ -1,3 +1,190 @@
+
+#### Comment Locations
+
+1. Functions should be preceeded by a short comment block explaining the relevant values (inputs, outputs) and an extremely brief description of the purpose of the function. This comment block should follow Doxygen styling, [linked here](https://www.doxygen.nl/manual/docblocks.html).
+
+```c title="Example"
+/**
+ * Sums two numbers
+ * @param A an integer
+ * @param B an integer
+ * @return The summation
+ */
+int sum(int A, int B) {
+    int c;
+    c = a + b;
+    return c;
+}
+```
+
+### White Space Rules
+
+1. The keywords ```if, while, for, switch,``` and ```return``` should be followed by a single space if there is more text on the same line.
+2. Unless specified by another whitespace rule, all operators should be surrounded on either side by a single space.
+3. Unary operators should not have a space between the operator and the operand.
+4. Structure pointer and member operators should have no whitespace on either side.
+5. Parentheses, braces, and brackets should not have an adjacent space on the interior side.
+6. Excluding the end of a line, separating commas should should be followed by one space.
+7. Semicolons should follow a preceeding statement without a space, but should always be followed by a space unless they end a line.
+8. Use whitespace to separate blocks of code from each other in a logical fashion. (William's note: to please my OCD use either 1 or 3 lines of separation depending on context, don't use two :) )
+
+#### Pointer (* and &) Operators
+
+1. Pointer operators should be written without a space on the operand side (i.e. ```type *name```).
+2. Pointer variables should be declared on a line separate from non-pointer variables of the same type, to avoid confusion between the two.
+
+#### Alignment Rules
+
+1. TODO: Consider whether to always left align the # of a preprocessor directive, regardless of the indentation of the line
+2. Align relevant blocks of code together (Don't be stupid) 
+    TODO: Make this nicer?
+
+#### Tab Rules
+
+1. The unicode tab character should not appear in files. Instead, the specified tab size is 4 spaces.
+
+### Module Rules
+
+#### Naming Conventions
+
+1. File names should consist only of lower case letters, numbers, and underscores.
+2. File names are recommended to be no longer than 16 characters. Ideally file names are specific and short, like the file itself.
+3. All module names should not overlap with those of the standard library.
+4. All files that contain a ```main()``` function should include "main" within the filename.
+
+#### Header Files
+
+1. Include preprocessor guards against multiple inclusion within all header files (#ifndef NAME_H)
+2. No variables should be declared within a header file, nor should any storage be allocated
+3. Header files should concern themselves strictly with the information that is needed to be known by other modules. (Everything else should be declared within the source file itself)
+
+#### Source Files
```

Comment:
Maybe somewhere in here add that there should typically be a header / source pair. One header for multiple source files is usually discouraged.